### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by lmoehlenbrock**

### CODE REVIEW

**Review by ovalle15 — COMMENTED** [review]
A few imports seem missing 
```
from pyspark.sql import SparkSession
from delta import configure_spark_with_delta_pip
```

File: notebooks/delta_table_upload_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+      "metadata": {
+        "id": "Lyy8pE_-WquW"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import labelspark as ls\n",
+        "import labelbox as lb\n",
+        "\n",
```
Comment (ovalle15) [inline notebooks/delta_table_upload_demo.ipynb:170]: Add imports here 
from pyspark.sql import SparkSession
from delta import configure_spark_with_delta_pip

File: notebooks/delta_table_upload_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+        "*   AWS_ACCCESS_KEY - Only required if connecting to AWS S3, the value should be your [AWS service account](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html) access key\n",
+        "*   AWS_SECRET_KEY - Only required if connecting to AWS S3, the value should be your [AWS service account](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html) secret key\n",
+        "\n"
+      ],
+      "metadata": {
+        "id": "y8b9Awq7YAfg"
+      }
+    },
+    {
+      "cell_type": "code",
```
Comment (ovalle15) [inline notebooks/delta_table_upload_demo.ipynb:249]: I would switch this table with the data table creation cell, I think it makes more sense to assume the customer is running this for the first time and don't have a table ready to test

File: notebooks/delta_table_upload_demo.ipynb [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+        "\n",
+        "gcs_credentials_file = '/content/{name of credentials .json file}'\n",
+        "\n",
+        "gcs_spark_config = {\n",
+        "                \"jars\": f\"{SPARK_HOME}/jars/gcs-connector-hadoop3-latest.jar\",\n",
+        "                \"credentials\": gcs_credentials_file\n",
+        "}\n",
+        "\n",
+        "#table_path can either be a local file, or a link to a third-party cloud storage service (e.g gs://{bucket_name}/{folder}/{table} or s3a://{bucket_name}/{folder}/{table})\n",
+        "table_path = \"\""
```
Comment (ovalle15) [inline notebooks/delta_table_upload_demo.ipynb:273]: Include the local path '/content/demo-table' here, if this cell is run after the data table creation step.

File: notebooks/delta_table_upload_demo.ipynb [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+        "\n"
+      ],
+      "metadata": {
+        "id": "y8b9Awq7YAfg"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from pyspark.sql import SparkSession\n",
```
Comment (ovalle15) [inline notebooks/delta_table_upload_demo.ipynb:251]: Remove imports here

File: notebooks/delta_table_upload_demo.ipynb [hunk 5]
```diff
@@ -0,0 +1,492 @@
...
+        "        builder.\\\n",
+        "        appName(\"pyspark-test\").\\\n",
+        "        config(\"spark.driver.extraClassPath\", f\"{SPARK_HOME}/jars/hadoop-aws-2.7.4.jar:{SPARK_HOME}/jars/aws-java-sdk-1.7.4.jar\").\\\n",
+        "        config(\"spark.executor.extraClassPath\", f\"{SPARK_HOME}/jars/hadoop-aws-2.7.4.jar:{SPARK_HOME}/jars/aws-java-sdk-1.7.4.jar\")\n",
+        "\n",
+        "spark = configure_spark_with_delta_pip(builder).getOrCreate()\n",
+        "spark._jsc.hadoopConfiguration().set(\"fs.s3a.access.key\", AWS_ACCESS_KEY)\n",
+        "spark._jsc.hadoopConfiguration().set(\"fs.s3a.secret.key\", AWS_SECRET_KEY)\n",
+        "\n",
+        "results = client.create_data_rows_from_delta_table(dataset_id=dataset.uid, project_id=project.uid, verbose=True, table_path=table_path, spark=spark)"
```
Comment (ovalle15) [inline notebooks/delta_table_upload_demo.ipynb:384]: Are you meant to run this here or in the next cell?

Comment (ovalle15) [inline notebooks/delta_table_upload_demo.ipynb:384]: Do we want results here to be overwritten by `results = client.create_data_rows_from_delta_table(dataset_id=dataset.uid, project_id=project.uid, verbose=True, table_path=table_path, spark_config=aws_spark_config) `in the next cell ?

File: notebooks/delta_table_upload_demo.ipynb [hunk 6]
```diff
@@ -0,0 +1,492 @@
...
+        "        config(\"spark.driver.extraClassPath\", f\"{SPARK_HOME}/jars/gcs-connector-hadoop3-latest.jar\").\\\n",
+        "        config(\"spark.executor.extraClassPath\", f\"{SPARK_HOME}/jars/gcs-connector-hadoop3-latest.jar\")\n",
+        "\n",
+        "spark = configure_spark_with_delta_pip(builder).getOrCreate()\n",
+        "spark.conf.set(\"spark.hadoop.fs.gs.auth.service.account.enable\", \"true\")\n",
+        "spark.conf.set(\"google.cloud.auth.type\", \"SERVICE_ACCOUNT_JSON_KEYFILE\")\n",
+        "spark.conf.set('google.cloud.auth.service.account.json.keyfile', credentials_file)\n",
+        "spark.conf.set(\"fs.AbstractFileSystem.gs.impl\", \"com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem\")\n",
+        "\n",
+        "results = client.create_data_rows_from_delta_table(dataset_id=dataset.uid, verbose=True, table_path=table_path, spark=spark)"
```
Comment (ovalle15) [inline notebooks/delta_table_upload_demo.ipynb:451]: I was not able to get any batch_to_project_results to pass successfully. 
`{'data_row_upload_results': 'Success',
 'batch_to_project_results': ValueError("{'status': 'FAILURE', 'results': ['', '', '', ''], 'errors': [{'global_key': 'data_row', 'error': 'Data Row not found'}, {'global_key': 'project_id', 'error': 'Data Row not found'}, {'global_key': 'annotations', 'error': 'Data Row not found'}, {'global_key': 'dataset_id', 'error': 'Data Row not found'}]}"),
 'annotation_upload_results': []}`

Comment (ovalle15) [reply notebooks/delta_table_upload_demo.ipynb:451]: I am assuming a batch needs to be created in the project but this is not done anywhere in the code.

File: notebooks/delta_table_upload_demo.ipynb (hunk 2 above)
Comment (lmoehlenbrock) [reply notebooks/delta_table_upload_demo.ipynb:249]: Good call, moved this cell and the two following cells just below the import cell

File: notebooks/delta_table_upload_demo.ipynb (hunk 5 above)
Comment (lmoehlenbrock) [reply notebooks/delta_table_upload_demo.ipynb:384]: Yeah, the intention here is that only one of the cells with client.create_data_rows_from_delta_table() will be run as this is showing that you can either run the function with a spark argument or spark_config argument

File: notebooks/delta_table_upload_demo.ipynb (hunk 1 above)
Comment (lmoehlenbrock) [reply notebooks/delta_table_upload_demo.ipynb:170]: done

File: notebooks/delta_table_upload_demo.ipynb (hunk 4 above)
Comment (lmoehlenbrock) [reply notebooks/delta_table_upload_demo.ipynb:251]: done

File: notebooks/delta_table_upload_demo.ipynb (hunk 3 above)
Comment (lmoehlenbrock) [reply notebooks/delta_table_upload_demo.ipynb:273]: done

**Comment by lmoehlenbrock** [comment]
I added a new commit addressing the comments

**Pull request merged**